File: Practice/AOJ/ALDS1/ALDS1_03/ALDS1_03B/alds1_03b.py
n, q = list(map(int, input().split()))
queue = [list(map(lambda x: int(x) if x.isnumeric() else x, input().split()))
         for i in range(n)]
queue.reverse()
c = 0
while True:
    if len(queue) == 0:
        break
    t = queue.pop()
    c += min(t[1], q)
    if t[1] <= q:
        print(t[0], c)
        continue
    t[1] -= q
    queue.insert(0, t)

"""
p2 180
p5 400
p1 450
p3 550
p4 800
"""

File: Practice/AOJ/ALDS1/ALDS1_03/ALDS1_03B/alds1_03b.rb
# 余った分は考えないらしい
n, q = gets.chomp.split.map(&:to_i)

queue = []

n.times do
  queue.push(gets.chomp.split.map{|x| x.match?(/^\d+$/) ? x.to_i : x})
end
# 入力された順に処理する
# FIFOは頭からいれてケツから出すという意識があるので
queue.reverse!
count = 0 # 経過時間
loop do
  break if queue.size.zero?
  # rubyではケツから取り出すのはpop
  t = queue.pop
  count += [t[1], q].min
  t[1] = [0, t[1]-q].max
  if t[1].zero?
    puts "#{t[0]} #{count}"
  else
    # rubyでは先頭にぶちこむのはunshift
    queue.unshift(t)
  end
end

